src/graphorm/graphorm.ts

```diff
@@ -293,13 +305,14 @@ export class GraphORM {
     fieldsByTypeName: { [p: string]: ResolveTree },
   ): GraphORMBuilder {
     const fields = Object.values(fieldsByTypeName);
-    const entityMetadata = ctx.con.getMetadata(
-      this.mappings?.[type]?.from || type,
-    );
+    const originalType = this.mappings?.[type]?.from ?? type;
+    const originalMetadata = ctx.con.getMetadata(originalType);
+    const tableName = originalMetadata.tableName;
```

Comment:
@idoshamun I encountered a bug where the expected entity was not used from the query (`from` property) since the table name used for the alias was from the `fromMetadata` hence, pulling the data from the reference entity instead of the view. Let me know if you would want this implemented differently.

Reply:
I probably missed this one. You can change it to `this.getMetadata(ctx, type)`

Reply:
Cool. Thank you for the confirmation.

Reply:
I requested to reverted this part to whatever there's in main now. Sorry I wasn't clear before 😅